src/MTGCore.Proxy/Controllers/HomeController.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Net.Http;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http.Extensions;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using MTGCore.Proxy.Models;
+using System.IO;
+
+namespace MTGCore.Proxy.Controllers
+{
+    public class HomeController : Controller
+    {
+        private readonly IHttpClientFactory _httpClientFactory;
+
+        public HomeController(IHttpClientFactory httpClientFactory)
+        {
+            _httpClientFactory = httpClientFactory;
+        }
+
+        [Route("{*anyRoute}")]
+        public async Task<IActionResult> cards(int id)
+
+        {
+            //https://localhost:44305/card/Details?id=397641
+
+            var path = Request.Path;
+
+            var hashedPath = CalculateMD5Hash(path);
+
+            var root = Environment.CurrentDirectory;
+            var fullPath = $@"{root}\Data\{hashedPath}.json";
+
+            if (!System.IO.File.Exists(fullPath))
+            {
+                var request = new HttpRequestMessage(HttpMethod.Get,
+                   "https://api.magicthegathering.io" + path);
```

Comment:
Given a production scenario, could we check not just on whether the path exists, but also when the file was last modified and if older than a certain date (I.E. 1 week) update it by calling the API?